async-nats/src/client.rs

```diff
@@ -45,6 +45,14 @@ impl From<tokio::sync::mpsc::error::SendError<Command>> for PublishError {
     }
 }
 
+pub struct MaxPayloadExceeded(String);
```

Comment:
This would work a bit better if it:

* Implemented `Debug`, `Display`, `Error`
* Had fields for the length of the body and the max payload size, with the string formatting done in the `Display` implementation

Reply:
I did it just to have an error in Draft.
Sure, your improvement is better, but that still makes this single error inconsistent with other errors, which have `ErrorKind`.